builders/ethereum/dev-env/thirdweb.md

```diff
@@ -110,184 +112,180 @@ thirdweb offers SDKs for a range of programming languages, such as React, React
-          // Calls the 'INSERT_NAME' function on your smart contract
-          // with 'INSERT_ARGUMENT' as the first argument
-          action={() => mutateAsync({ args: ['INSERT_ARGUMENT'] })}
-        >
-          Send Transaction
-        </Web3Button>
-      );
-    }
-    ```
+const number = await readContract({
+      contract: contract,
+      method: 'number',
+      params: [],
+    });
+```
 
 ### Connect Wallet {: #connect-wallet }
 
-There are a couple of ways that you can create a custom [connect wallet](https://portal.thirdweb.com/react/connecting-wallets){target=\_blank} experience. You can use the [`ConnectWallet` component](https://portal.thirdweb.com/react/connecting-wallets#using-the-connect-wallet-button){target=\_blank} or, for a more customizable approach, you can use the [`useConnect` hook](https://portal.thirdweb.com/react/connecting-wallets#using-hooks){target=\_blank}.
-
-The following example will show you how to use the `ConnectWallet` component. To go this route, you will need to specify the supported wallets and pass them to your provider.
-
-```javascript
-import {
-  ThirdwebProvider,
-  metamaskWallet,
-  coinbaseWallet,
-  walletConnectV1,
-  walletConnect,
-  safeWallet,
-  paperWallet,
-} from '@thirdweb-dev/react';
-
-function MyApp() {
-  return (
-    <ThirdwebProvider
-      supportedWallets={[
-        metamaskWallet(),
-        coinbaseWallet(),
-        walletConnect({
-          projectId: 'INSERT_YOUR_PROJECT_ID', // optional
-        }),
-        walletConnectV1(),
-        safeWallet(),
-        paperWallet({
-          clientId: 'INSERT_YOUR_CLIENT_ID', // required
-        }),
-      ]}
-      activeChain={Moonbeam}
-    >
-      <App />
-    </ThirdwebProvider>
-  );
-}
-```
+Next, you can customize the [Connect Button](https://portal.thirdweb.com/typescript/v5/react/components/ConnectButton){target=\_blank} to tailor it our desired wallets. The Connect Button accepts an optional  `wallets` parameter with an array of wallets. You can add or remove wallets from the wallets array to change the options available to users. ThirdWeb also offers a [ConnectButton Playground](https://thirdweb.com/dashboard/connect/playground) to customize and view changes in real-time given the high degree of flexibility offered by the button. The below snippet will also include logic to display the counter value and the increment button.
 
-Next, you'll need to add in a connect wallet button to prompt end-users to login with any of the above supported wallets.
+```typescript title="App.tsx"
+    --8<-- 'code/builders/ethereum/dev-env/thirdweb/App.tsx:2:2'
+    --8<-- 'code/builders/ethereum/dev-env/thirdweb/App.tsx:7:7'
 
-```javascript
-import { ConnectWallet } from '@thirdweb-dev/react';
+    --8<-- 'code/builders/ethereum/dev-env/thirdweb/App.tsx:9:15'
 
-function App() {
-  return <ConnectWallet />;
-}
+    --8<-- 'code/builders/ethereum/dev-env/thirdweb/App.tsx:87:105'
```

Comment:
Whichever one of these snippets is the one where the wallet options get added: there is a Header component in the JSX that we never create or import, so it's throwing an error